mitodl/ocw-hugo-themes
# chore(dep): reorganize dependencies

### What are the relevant tickets?
Part of https://github.com/mitodl/ocw-hugo-themes/issues/1296

### Description (What does it do?)

This PR:
- Reorganizes dependencies into `dependencies` and `devDependencies`.
- Removes unused dependencies.

This is the first step in the plan laid out in https://github.com/mitodl/ocw-hugo-themes/issues/1296.

The following dependencies were removed because **no direct usage** was found.

```json
{
    "@types/lodash.flatten": "^4.4.6",
    "@types/lodash.toarray": "^4.4.6",
    "lodash.debounce": "^4.0.8",
    "lodash.flatten": "^4.4.0",
    "lodash.toarray": "^4.4.0",
    "archiver": "^5.0.0",
    "bodybuilder": "^2.2.21",
    "cli-progress": "^3.8.2",
    "fancy-log": "^1.3.2",
    "npm-run-all": "^4.1.5",
    "raf": "^3.4.1",
    "shelljs": "^0.8.4",
    "spinkit": "^2.0.1",
    "tmp": "^0.2.1",
    "walk": "^2.3.14",
    "react-test-renderer": "^16.13.1",
    "gulp": "^4.0.2",
    "react-hot-loader": "^4.12.17",
    "plugin-error": "^1.0.1",
    "style-loader": "^1.0.0",
    "url-loader": "^3.0.0",
    "babel-loader": "^8.2.5",
    "exports-loader": "^0.7.0",
    "expose-loader": "^4.0.0"
}
```


### How can this be tested?

1. Grab a course and [www](https://github.mit.edu/ocw-content-rc/ocw-www) content. You may use [ht-test-2024](https://github.mit.edu/ocw-content-rc/ht-test-2024) (transcripts don't exist in this one) or [18.03-spring-2010](https://github.mit.edu/ocw-content-rc/18.03-spring-2010).
2. Place the content in your local content directory.
3. Navigate to your local `ocw-hugo-theme` directory.
4. Checkout the branch `hussaintaj/1296-reorganize-deps`.
5. Delete the `node_modules` folder.
6. Clean cache.
    ```sh
    yarn cache clean
    ```
7. Install dependencies.
   ```sh
   yarn
   ```
8. Run a course.
    ```sh
    yarn start course 18.03-spring-2010
    ```
9. Open [http://localhost:3000](http://localhost:3000)
10. Expect to see everything working. This includes:
     - Menus, toggles, and other interactive elements.
     - Videos, transcripts, PDF reader, etc.
     - Image Gallery, quizzes.
11. Build the course for offline use:
     ```sh
     yarn build ~/mitodl/ocw-content/<id> ~/mitodl/ocw-hugo-projects/ocw-course-v2/config-offline.yaml
     ```
12. Open the file `~/mitodl/ocw-content/<id>/dist/index.html` or wherever you stored the build.
13. Repeat step 10.
14. Run www.
     ```sh
     yarn start www
     ```
15. Expect everything to be working. This includes:
     - Menus, sliders, and other interactive elements.
     - Search. (You might need an extension that permits [CORS](https://addons.mozilla.org/en-US/firefox/addon/cors-everywhere/))
16. Build www for offline use.
     ```sh
     yarn build ~/mitodl/ocw-content/ocw-www ~/mitodl/ocw-hugo-projects/ocw-www/config-offline.yaml
     ```
17. Open the file `~/mitodl/ocw-content/ocw-www/dist/index.html` or wherever you stored the build.
18. Expect the UI to work.
19. Expect our script utilities to work:
     ```sh
     yarn lint
     subset-fonts
     prebuild
     build:githash
     build:webpack
     build:webpack:dev
     webpack:serve
     webpack:stats
     webpack:analyze
     fmt
     scss_lint
     test:watch
     test:e2e
     test:e2e:jest
     typecheck
     ```
     > subset-fonts requires `brew install fonttools`. I'm unsure why `build:hugo` exists but I believe we no longer use it.
